model/src/main/kotlin/PackageCuration.kt

```diff
@@ -37,18 +37,10 @@ data class PackageCuration(
     val data: PackageCurationData
 ) {
     /**
-     * Apply the curation [data] to the provided package.
+     * Apply the curation [data] to the provided package. The caller has to take care that [data] is actually applicable
+     * to the [curatedPackage].
      *
      * @see [PackageCurationData.apply]
      */
-    fun apply(curatedPackage: CuratedPackage): CuratedPackage {
-        if (!id.matches(curatedPackage.pkg.id)) {
-            throw IllegalArgumentException(
-                "Package curation identifier '${id.toCoordinates()}' does not match " +
-                        "package identifier '${curatedPackage.pkg.id.toCoordinates()}'."
-            )
-        }
-
-        return data.apply(curatedPackage)
-    }
+    fun apply(curatedPackage: CuratedPackage) = data.apply(curatedPackage)
```

Comment:
You wouldn't have to remove this safety net if the version range was supported in `Identifier.matches()`. :)

The KDoc for `id` of this class should explain that Ivy matchers are supported.